### PR TITLE
[UserBundle] Migration script

### PR DESCRIPTION
**Pull request by michalmarcinkowski**
Initial migration script for SyliusUserBundle. It creates customers based on users and guest orders and also preserves addresses, groups and orders by associating them with appropriate customers.

TODO:
 - [x] Permissions migration
 - [x] Test on real data

#### UPDATE

Permissions should be initialized using command ```sylius:rbac:initialize```

### CODE REVIEW

**Comment by nicolasricci** [comment]
thanks @michalmarcinkowski 
I will try to give it a try tonight but I don't have any Sylius website in production at the moment so not sure the few tests I have recorded in my database will cover all user cases.

File: app/migrations/Version20150531210410.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        $this->addSql('ALTER TABLE sylius_promotion_coupon CHANGE per_user_usage_limit per_customer_usage_limit INT DEFAULT NULL');
+
+        // migrate orders
+        $this->addSql('ALTER TABLE sylius_order DROP FOREIGN KEY FK_6196A1F9A76ED395');
+        $this->addSql('DROP INDEX IDX_6196A1F9A76ED395 ON sylius_order');
+        $this->addSql('ALTER TABLE sylius_order ADD customer_id INT DEFAULT NULL');
+        $this->addSql('ALTER TABLE sylius_order ADD CONSTRAINT FK_6196A1F99395C3F3 FOREIGN KEY (customer_id) REFERENCES sylius_customer (id)');
+        $this->addSql('CREATE INDEX IDX_6196A1F99395C3F3 ON sylius_order (customer_id)');
+        // create customers based on guest checkouts
+        $this->addSql('INSERT INTO sylius_customer (email, email_canonical, created_at) SELECT sylius_order.email, sylius_order.email, sylius_order.created_at FROM sylius_order LEFT JOIN sylius_customer ON sylius_order.email = sylius_customer.email WHERE sylius_customer.email is NULL');
```
Comment (nicolasricci) [inline app/migrations/Version20150531210410.php]: I guess you meant IS NOT NULL to retrieve only user with associated email addresses

**Comment by nicolasricci** [comment]
Apart from the line note above everything looks fine for me.

File: app/migrations/Version20150531210410.php (hunk 1 above)
Comment (michalmarcinkowski) [inline app/migrations/Version20150531210410.php]: That is correct, I meant IS NULL. Here I create customers which are not users (not registered) and did a guest checkout. So I want to select all guest orders for which there is no customer with corresponding email.

Comment (nicolasricci) [inline app/migrations/Version20150531210410.php]: The problem is that the email address cannot be null in sylius_customer table so you will have the following violation if you keep it as it is

```
SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'email' cannot be null  
```

Comment (michalmarcinkowski) [inline app/migrations/Version20150531210410.php]: But I use sylius_order.email to create customer. I think the missing check here is 'where sylius_order.email is not null', will add it tomorrow :)

**Comment by pjedrzejewski** [comment]
:+1:

**Pull request merged**